### ISSUE TITLE
Fix Epf config for #91, #135, #155 and #165

### ISSUE THREAD

**Issue by rw00** [dubreuia/intellij-plugin-save-actions#178]
Unaware, I added random properties to EpfStorage class in my pull requests for those issues. 
I didn't have Eclipse installation on my machine and to be honest I didn't think about it much

Here's a sample Eclipse Preferences data: 

```
sp_cleanup.add_default_serial_version_id=false
sp_cleanup.add_generated_serial_version_id=false
sp_cleanup.add_missing_annotations=true
sp_cleanup.add_missing_deprecated_annotations=false
sp_cleanup.add_missing_methods=false
sp_cleanup.add_missing_nls_tags=false
sp_cleanup.add_missing_override_annotations=true
sp_cleanup.add_missing_override_annotations_interface_methods=false
sp_cleanup.add_serial_version_id=false
sp_cleanup.always_use_blocks=false
sp_cleanup.always_use_parentheses_in_expressions=false
sp_cleanup.always_use_this_for_non_static_field_access=false
sp_cleanup.always_use_this_for_non_static_method_access=false
sp_cleanup.convert_functional_interfaces=false
sp_cleanup.convert_to_enhanced_for_loop=false
sp_cleanup.correct_indentation=false
sp_cleanup.format_source_code=false
sp_cleanup.format_source_code_changes_only=false
sp_cleanup.insert_inferred_type_arguments=false
sp_cleanup.make_local_variable_final=false
sp_cleanup.make_parameters_final=false
sp_cleanup.make_private_fields_final=false
sp_cleanup.make_type_abstract_if_missing_method=false
sp_cleanup.make_variable_declarations_final=false
sp_cleanup.never_use_blocks=false
sp_cleanup.never_use_parentheses_in_expressions=false
sp_cleanup.on_save_use_additional_actions=true
sp_cleanup.organize_imports=false
sp_cleanup.qualify_static_field_accesses_with_declaring_class=false
sp_cleanup.qualify_static_member_accesses_through_instances_with_declaring_class=false
sp_cleanup.qualify_static_member_accesses_through_subtypes_with_declaring_class=false
sp_cleanup.qualify_static_member_accesses_with_declaring_class=false
sp_cleanup.qualify_static_method_accesses_with_declaring_class=false
sp_cleanup.remove_private_constructors=false
sp_cleanup.remove_redundant_type_arguments=true
sp_cleanup.remove_trailing_whitespaces=false
sp_cleanup.remove_trailing_whitespaces_all=false
sp_cleanup.remove_trailing_whitespaces_ignore_empty=false
sp_cleanup.remove_unnecessary_casts=false
sp_cleanup.remove_unnecessary_nls_tags=false
sp_cleanup.remove_unused_imports=true
sp_cleanup.remove_unused_local_variables=false
sp_cleanup.remove_unused_private_fields=false
sp_cleanup.remove_unused_private_members=false
sp_cleanup.remove_unused_private_methods=false
sp_cleanup.remove_unused_private_types=false
sp_cleanup.sort_members=false
sp_cleanup.sort_members_all=false
sp_cleanup.use_anonymous_class_creation=false
sp_cleanup.use_blocks=false
sp_cleanup.use_blocks_only_for_return_and_throw=false
sp_cleanup.use_lambda=false
sp_cleanup.use_parentheses_in_expressions=false
sp_cleanup.use_this_for_non_static_field_access=false
sp_cleanup.use_this_for_non_static_field_access_only_if_necessary=false
sp_cleanup.use_this_for_non_static_method_access=false
sp_cleanup.use_this_for_non_static_method_access_only_if_necessary=false
```

**Comment by dubreuia** [comment]
> Unaware, I added random properties to EpfStorage class in my pull requests for those issues

:o 

If some do no match, I prefer having nothing than a property that do not exists in Eclipse. Also, this code `com.dubreuia.model.EpfStorage` isn't properly tested.

**Comment by dubreuia** [comment]
Hey @markiewb I'm fixing this issue, do you remember what is the difference between those two files: `src/test/resources/example.epf` and `src/test/resources/org.eclipse.jdt.ui.prefs`

**Comment by dubreuia** [comment]
@rw00 @markiewb Do you know a place where I can find the eclipse documentation on epf files? I'd like to have all the possible keys for the `sp_cleanup` prefix.

**Comment by rw00** [comment]
@dubreuia  I think the ones I included in this issue are all. 
In Eclipse, go to Preferences < Java < Editor < Save Actions, modify some settings and finally save. 

Check the file which stores these settings under: 
`<your-eclipse-workspace>/.metadata/.plugins/org.eclipse.core.runtime/.settings/org.eclipse.jdt.ui.prefs`

I couldn't find any documentation for these settings

**Comment by dubreuia** [comment]
Ok thanks, I'm refactoring + testing the class

**Comment by markiewb** [comment]
@dubreuia: @rw00 explained it correct. Initially I did the same as him. Changed the setting in Eclipse and watch for the change in the preference file. 

AFAIK there is no official documentation of the available keys. Every setting in Eclipse has 1 or more corresponding setting entries in org.eclipse.jdt.ui.prefs. The epf-file is a xml-file based on the key-value pairs of this file, which is supported by the Workspace mechanic plugin https://marketplace.eclipse.org/content/workspace-mechanic
